### ISSUE TITLE
Port raiser.c to 3.1

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4247]
|<img alt="loewis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| loewis reported|
|-|-|
|Trac ID|trac#4247|
|Type|enhancement|
|Created|2010-01-30 18:06:50Z|
|Branch|https://github.com/twisted/twisted/tree/raiser-py3k-4247|

Porting raiser.c is fairly easy - just run raiser.pyx through a recent version of Cython. Patch attached. The test still passes in 2.5.

Attachments:

* [raiser.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bb7/bb7aa5a065e5d980ec6e5e54790538509980b9e3/26d2da611b58376ae0570450754d636392c2df00.diff) (57357 bytes) - added by loewis on 2010-01-30 18:07:19Z - 
* [_epoll.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bb7/bb7aa5a065e5d980ec6e5e54790538509980b9e3/b620475a7dab3e6a65336ceca8c31c964c933819.diff) (138805 bytes) - added by loewis on 2010-01-30 18:18:53Z - 
* [4247.misc](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bb7/bb7aa5a065e5d980ec6e5e54790538509980b9e3/c1f69e7dd2598cda1bc0517e1aab7c6fedfbbf50.misc) (429 bytes) - added by loewis on 2010-01-31 08:10:26Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4247 4247
type__enhancement enhancement
reporter__loewis loewis
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_raiser_py3k_4247 branches/raiser-py3k-4247
branch_author__loewis__thijs loewis, thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__py3k py3k
time__1264874810000000 1264874810000000
changetime__1300125076000000 1300125076000000
version__None None
owner__loewis loewis
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="loewis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|loewis commented|
|-|-|

Likewise for _epoll.

**Comment by twisted-trac** [comment]
|<img alt="loewis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|loewis removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra set owner to loewis|
|-|-|

Looks like this is blocked by [#4246](https://github.com/twisted/twisted/issues/4246) that wants tries to fix some other py3k stuff.. If anyone works on this ticket in the future, make sure to run it through the latest version of Cython (0.12.1 at the time of this writing) cause this patch uses an slightly outdated version of Cython.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Replying to [thijs](#note_4):
> Looks like this is blocked by [#4246](https://github.com/twisted/twisted/issues/4246) that wants tries to fix some other py3k stuff..

That should be [#4244](https://github.com/twisted/twisted/issues/4244).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [30892]) Branching to 'raiser-py3k-4247'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

(In [30893]) refresh pyx output files with cython 0.14.1, refs [#4247](https://github.com/twisted/twisted/issues/4247)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [30894]) add news file, refs [#4247](https://github.com/twisted/twisted/issues/4247)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra removed owner|
|-|-|

I updated the files with Cython 0.14.1 and all tests passed here with python 2.7.1. I don't think this necessarily should be blocked by [#4244](https://github.com/twisted/twisted/issues/4244) (besides adding py3k support it could potentially fix/improve other stuff) so putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)<a name="note_10"></a>|@washort set owner to loewis|
|-|-|

Tests pass on debian64-py2.4-select and debian-easy-py2.5-epoll. Since this was just a rebuild of pyx files, I'm approving this.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [31032]) Merge raiser-py3k-4247: Updated raiser.c and _epoll.c with Cython 0.14.1 to support Python 3.

Author: loewis, thijs
Reviewer: washort
Fixes: [#4247](https://github.com/twisted/twisted/issues/4247)
